src/main/webapp/app/exercises/shared/exercise-headers/header-exercise-page-with-details.component.ts

```diff
@@ -72,4 +90,32 @@ export class HeaderExercisePageWithDetailsComponent implements OnChanges {
             this.icon = getIcon(exerciseType) as IconProp;
         }
     }
+
+    private setIsNextDueDateExamMode() {
+        this.isNextDueDate = [false, false];
+        const now = dayjs();
+        if (now.isBefore(this.exam?.endDate)) {
+            this.isNextDueDate[0] = true;
+        } else if (now.isBefore(this.exam?.publishResultsDate)) {
+            this.isNextDueDate[1] = true;
+        }
+    }
+
+    private setIsNextDueDateCourseMode() {
```

Comment:
Could you please add comments. Thank you!

Reply:
Additionally, there [some branches](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS3347-42/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/exercise-headers/header-exercise-page-with-details.component.ts.html) here are not yet covered. If it is not too much work, could you add client tests for those?